### DISCUSSION TITLE
Google Chrome Extended Stable?

### DISCUSSION THREAD

**Discussion by bmitchell76** [EUCPilots/evergreen-module#326] [Ideas]
Remember when Chrome was on a six-week release cycle?  4 - 5 updates a month later...  Any chance there is a feed to add the Google Chrome Extended Stable version?

http://chromereleases.googleblog.com/2022/05/extended-stable-channel-update-for.html

**Comment by aaronparker** [comment]
Like Edge, Extended Stable isn't a channel or a specific release, it's a policy that you set. Neither the Edge or Chrome update feeds specifically call out the Extended Stable version. There might be something out there, but I haven't found it yet.

**Reply by bmitchell76** [reply]
I set the ADMX to Extended and did an update and this is what I captured with Fiddler.  It doesn't return the MSI but does the release channel version.

https://update.googleapis.com/service/update2?cup2key=10:3539294517&cup2hreq=9b86f646da4ced4692e546f1dbf2ae8a75537b7a17232480f6fbeb474a64f711

![Chrome ExtendedStable](https://user-images.githubusercontent.com/39991783/168201957-6dc026a6-73f9-42be-a32c-45d77d0d3cc2.jpg)

**Reply by bmitchell76** [reply]
Interestingly, I can't find an offline install for 100.0.4896.160;  you'd think that if an install were called "Extended Stable," the install would be readily available.  As far as I can tell, that version is limited to if Chrome auto-update.

**Reply by aaronparker** [reply]
I've been doing a bit of testing by making calls to that API with no luck so far.

**Comment by aaronparker** [comment]
Here's how to query the API, but as you've pointed out, downloading that version is a challenge.

```powershell
$headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
$headers.Add("X-Goog-Update-AppId", "{8A69D345-D564-463C-AFF1-A69D9E530F96}")
$headers.Add("X-Goog-Update-Interactivity", "fg")
$headers.Add("X-Goog-Update-Updater", "Omaha-1.3.36.132")
$headers.Add("X-Last-HR", "0x0")
$headers.Add("X-Last-HTTP-Status-Code", "0")
$headers.Add("X-Retry-Count", "0")
$headers.Add("Content-Type", "text/plain")

$body = "<?xml version=`"1.0`" encoding=`"UTF-8`"?><request protocol=`"3.0`" updater=`"Omaha`" updaterversion=`"1.3.36.132`" shell_version=`"1.3.36.131`" ismachine=`"1`" sessionid=`"{4F8FA38F-2097-48F1-B4DE-9D8DCA2DA207}`" installsource=`"update3web-ondemand`" requestid=`"{C0798A5C-CE46-41FB-9CDD-0511C4494556}`" dedup=`"cr`" domainjoined=`"1`"><hw physmemory=`"3`" sse=`"1`" sse2=`"1`" sse3=`"1`" ssse3=`"1`" sse41=`"1`" sse42=`"1`" avx=`"1`"/><os platform=`"win`" version=`"10.0.19044.1706`" sp=`"`" arch=`"x64`"/><app appid=`"{8A69D345-D564-463C-AFF1-A69D9E530F96}`" version=`"90.0.4951.67`" nextversion=`"`" ap=`"x64-stable`" lang=`"`" brand=`"GCEA`" client=`"`" installage=`"0`" installdate=`"5607`" cohort=`"1:gu/i19:`" cohortname=`"Stable Installs &amp; Version Pins`"><updatecheck release_channel=`"extended`"/><ping active=`"1`" ad=`"5612`" rd=`"5612`" ping_freshness=`"{AA0DF675-8107-4C72-A434-FA1F768C530F}`"/></app></request>"

$params = @{
    Uri     = "https://update.googleapis.com/service/update2?cup2key=11:2p4siNgnztkc2kPrQCYESLPFlWu2xWQ0W7dJBQToYcI&cup2hreq=b9c0d5fc243dccbf9dcc88e41725b1cacad398e47d287cfbd9e9a66f3c5cefa4"
    Method  = "POST"
    Headers = $headers
    Body    = $body
}
$r = Invoke-RestMethod @params
$r.response.app.updatecheck.manifest.packages.package
```

**Reply by bmitchell76** [reply]
I'm coming up blank finding the download.  They don't want you to update Chrome outside the auto-update feature.

**Comment by psomeren** [comment]
"Extended" versions are also "Stable" versions. Please, compare the results from Chromium Dash URL's

https://versionhistory.googleapis.com/v1/chrome/platforms/all/channels/all/versions/all/releases
This URL shows (in JSON) all platforms, all channels, all versions and all releases. When you search for the latest "Extended Channel" version, you will also find that same version as "Stable Channel".

If you want to narrow it down and show "Stable Channel" versions only, modify the above URL like this:

https://versionhistory.googleapis.com/v1/chrome/platforms/win/channels/stable/versions/all/releases
Now, the URL only shows Platform Windows, Stable Channel, all versions and and releases. The first one is usually the latest version when you download it at https://dl.google.com/dl/chrome/install/googlechromestandaloneenterprise64.msi
